pkg/rtc/mediatrack.go

```diff
@@ -204,6 +206,7 @@ func (t *MediaTrack) AddReceiver(receiver *webrtc.RTPReceiver, track *webrtc.Tra
 		}
 	})
 
+	t.lock.Lock()
```

Comment:
Is there an alternative option here? The statement `Lock data, don't lock code` always rings in my head :-) Although I violate that a lot, but a lock specifically for a small section of code makes me feel odd.

Reply:
yeah, a dedicated lock for this code looks odd now, I import this lock maybe it can be used later...
I have thought best way is protect this section with MediaTrackReceiver's lock, something like MediaTrackReceiver.LoadOrSetReceiver(f func () TrackReceiver {}). but it still a little odd for me..

Reply:
Sounds good Jie. Will file an issue for this to look at later.